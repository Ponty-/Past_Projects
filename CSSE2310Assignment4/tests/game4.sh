#!/bin/bash

if [ $# != 4 ]
then
    echo "Usage: $0 binarydir port resultdir correct_dir" 
    exit 1
fi



bindir=$1
port=$2
res=$3
right=$4
name=$0

expected=0

if [ ! -f $bindir/client499 ]
then
    echo "$bindir/client499 is missing"
    exit 1
fi

if [ ! -f $bindir/serv499 ]
then
    echo "$bindir/serv499 is missing"
    exit 1
fi



$bindir/serv499 $port '%s%s%s%s%s' 4.deck &
spid=$!
$bindir/client499 zzzzzzzz gg $port > $res/$name.a.out 2>$res/$name.a.err < ax &
pid1=$!
$bindir/client499 123456 gg $port > $res/$name.b.out 2>$res/$name.b.err < bx &
pid2=$!
$bindir/client499 CCC gg $port > $res/$name.c.out 2>$res/$name.c.err < cx &
pid3=$!
$bindir/client499 dddd gg $port > $res/$name.d.out 2>$res/$name.d.err < dx &
pid4=$!

wait $pid1 $pid2 $pid3 $pid4 2> /dev/null

$bindir/client499 zzzzA game $port > $res/$name.a.2.out 2>$res/$name.a.2.err < ax &
pid1=$!
$bindir/client499 123B game $port > $res/$name.b.2.out 2>$res/$name.b.2.err < bx &
pid2=$!
$bindir/client499 C game $port > $res/$name.c.2.out 2>$res/$name.c.2.err < cx &
pid3=$!
$bindir/client499 D game $port > $res/$name.d.2.out 2>$res/$name.d.2.err < dx &
pid4=$!

wait $pid1 $pid2 $pid3 $pid4 2> /dev/null



kill $spid 2> /dev/null
wait $spid 2> /dev/null

#if [ $result != $expected ]
#then
#    echo "Expected $expected got $result"
#    exit 1
#fi

for v in a b c d
do

diff -q $res/$name.$v.2.out $right/$name.$v.2.out > /dev/null
if [ $? != 0 ]
then
   echo "Stdout mismatch for client $v"
   echo "diff $res/$name.$v.2.out $right/$name.$v.2.out"
   exit 2
fi

diff -q $res/$name.$v.2.err $right/$name.$v.2.err > /dev/null
if [ $? != 0 ]
then
   echo "Stderr mismatch for client $v"
   echo "diff $res/$name.$v.2.err $right/$name.$v.2.err"
   exit 3
fi

done

echo OK
exit 0

