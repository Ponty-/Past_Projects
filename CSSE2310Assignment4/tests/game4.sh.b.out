Info: %s%s%s%s%s
Info: Team1: 123456, dddd
Info: Team2: CCC, zzzzzzzz
S: A K J 8 5 3 2
C: J 8 5
D:
H: Q 7 3
Bid> Info: CCC passes
Info: dddd bids 8S
Info: zzzzzzzz passes
[8S] - Bid (or pass)> Info: dddd plays TS
Info: zzzzzzzz plays 9S
S: A K J 8 5 3 2
C: J 8 5
D:
H: Q 7 3
[S] play> Info: CCC plays 7S
Info: 123456 won
S: K J 8 5 3 2
C: J 8 5
D:
H: Q 7 3
Lead> Info: CCC plays QS
Info: dddd plays 6S
Info: zzzzzzzz plays 4S
Info: 123456 won
S: J 8 5 3 2
C: J 8 5
D:
H: Q 7 3
Lead> Info: CCC plays AH
Info: dddd plays 4H
Info: zzzzzzzz plays 5H
Info: CCC won
Info: CCC plays QC
Info: dddd plays 2C
Info: zzzzzzzz plays 4C
S: J 8 5 3 2
C: J 8 5
D:
H: 7 3
[C] play> S: J 8 5 3 2
C: J 8 5
D:
H: 7 3
[C] play> Info: CCC won
Info: CCC plays KD
Info: dddd plays 3D
Info: zzzzzzzz plays 2D
S: J 8 5 3 2
C: J 8
D:
H: 7 3
[D] play> Info: 123456 won
S: 8 5 3 2
C: J 8
D:
H: 7 3
Lead> Info: CCC plays TH
Info: dddd plays 9H
Info: zzzzzzzz plays 8H
Info: CCC won
Info: CCC plays 7C
Info: dddd plays AC
Info: zzzzzzzz plays 9C
S: 8 5 3 2
C: J 8
D:
H: 3
[C] play> Info: dddd won
Info: dddd plays KH
Info: zzzzzzzz plays JH
S: 8 5 3 2
C: J
D:
H: 3
[H] play> Info: CCC plays 2H
Info: dddd won
Info: dddd plays TC
Info: zzzzzzzz plays KC
S: 8 5 3 2
C: J
D:
H:
[C] play> Info: CCC plays 3C
Info: zzzzzzzz won
Info: zzzzzzzz plays AD
S: 8 5 3 2
C:
D:
H:
[D] play> Info: CCC plays 5D
Info: dddd plays JD
Info: 123456 won
S: 8 5 3
C:
D:
H:
Lead> Info: CCC plays 7D
Info: dddd plays 6C
Info: zzzzzzzz plays QD
Info: 123456 won
S: 5 3
C:
D:
H:
Lead> Info: CCC plays TD
Info: dddd plays 6D
Info: zzzzzzzz plays 4D
Info: 123456 won
S: 5
C:
D:
H:
Lead> Info: CCC plays 6H
Info: dddd plays 9D
Info: zzzzzzzz plays 8D
Info: 123456 won
Info: Team 1=220, Team 2=0
S: A K J 8 5 3 2
C: J T 5
D:
H: Q 7 3
Bid> Bid> Info: CCC bids 6H
Info: dddd passes
Info: zzzzzzzz passes
[6H] - Bid (or pass)> [6H] - Bid (or pass)> Info: CCC plays AH
Info: dddd plays KH
Info: zzzzzzzz plays 5H
S: A K J 8 5 3 2
C: J T 5
D:
H: Q 7 3
[H] play> Info: CCC won
Info: CCC plays KD
Info: dddd plays 3D
Info: zzzzzzzz plays AD
S: A K J 8 5 3 2
C: J T 5
D:
H: Q 7
[D] play> Info: zzzzzzzz won
Info: zzzzzzzz plays KC
S: A K J 8 5 3
C: J T 5
D:
H: Q 7
[C] play> Info: CCC plays 3C
Info: dddd plays 2C
Info: zzzzzzzz won
Info: zzzzzzzz plays QD
S: A K J 8 5 3
C: J T
D:
H: Q 7
[D] play> Info: CCC plays 5D
Info: dddd plays 6D
Info: zzzzzzzz won
Info: zzzzzzzz plays 9S
S: A K J 8 5 3
C: J
D:
H: Q 7
[S] play> Info: CCC plays 7S
Info: dddd plays 6S
Info: 123456 won
S: K J 8 5 3
C: J
D:
H: Q 7
Lead> Info: CCC plays QS
Info: dddd plays TS
Info: zzzzzzzz plays 4S
Info: 123456 won
S: J 8 5 3
C: J
D:
H: Q 7
Lead> Info: CCC plays TH
Info: dddd plays 6C
Info: zzzzzzzz plays 2D
Info: CCC won
Info: CCC plays QC
Info: dddd plays AC
Info: zzzzzzzz plays 4C
S: J 8 5
C: J
D:
H: Q 7
[C] play> Info: dddd won
Info: dddd plays 4H
Info: zzzzzzzz plays 8H
S: J 8 5
C:
D:
H: Q 7
[H] play> Info: CCC plays 2H
Info: zzzzzzzz won
Info: zzzzzzzz plays JH
S: J 8 5
C:
D:
H: Q
[H] play> Info: CCC plays 6H
Info: dddd plays 9H
Info: 123456 won
S: J 8 5
C:
D:
H:
Lead> Info: CCC plays 7D
Info: dddd plays 9D
Info: zzzzzzzz plays 4D
Info: 123456 won
S: J 8
C:
D:
H:
Lead> Info: CCC plays 9C
Info: dddd plays 8C
Info: zzzzzzzz plays 7C
Info: 123456 won
S: J
C:
D:
H:
Lead> Info: CCC plays TD
Info: dddd plays JD
Info: zzzzzzzz plays 8D
Info: 123456 won
Info: Team 1=220, Team 2=150
S: Q 7
C: Q 7 3
D: K T 7 5
H: A T 6 2
Bid> Info: CCC bids 7H
Info: dddd passes
Info: zzzzzzzz passes
[7H] - Bid (or pass)> Info: CCC plays AC
Info: dddd plays KC
Info: zzzzzzzz plays 5C
S: Q 7
C: Q 7 3
D: K T 7 5
H: A T 6 2
[C] play> Info: CCC won
Info: CCC plays TC
Info: dddd plays 4C
Info: zzzzzzzz plays 8C
S: Q 7
C: Q 7
D: K T 7 5
H: A T 6 2
[C] play> Info: CCC won
Info: CCC plays KH
Info: dddd plays 5H
Info: zzzzzzzz plays 3H
S: Q 7
C: Q
D: K T 7 5
H: A T 6 2
[H] play> Info: CCC won
Info: CCC plays JD
Info: dddd plays 2D
Info: zzzzzzzz plays 2S
S: Q 7
C: Q
D: K T 7 5
H: A T 6
[D] play> Info: CCC won
Info: CCC plays 9D
Info: dddd plays 4D
Info: zzzzzzzz plays 3S
S: Q 7
C: Q
D: K T 7
H: A T 6
[D] play> Info: CCC won
Info: CCC plays 9H
Info: dddd plays JH
Info: zzzzzzzz plays 7H
S: Q 7
C: Q
D: K T
H: A T 6
[H] play> Info: 123456 won
S: Q 7
C: Q
D: K T
H: T 6
Lead> Info: CCC plays 2C
Info: dddd plays 9C
Info: zzzzzzzz plays JC
Info: 123456 won
S: Q 7
C:
D: K T
H: T 6
Lead> Info: CCC plays 4H
Info: dddd plays 8H
Info: zzzzzzzz plays QH
Info: zzzzzzzz won
Info: zzzzzzzz plays 5S
S: Q 7
C:
D: K T
H: T
[S] play> Info: CCC plays 6S
Info: dddd plays 4S
Info: 123456 won
S: 7
C:
D: K T
H: T
Lead> Info: CCC plays TS
Info: dddd plays 9S
Info: zzzzzzzz plays 8S
Info: CCC won
Info: CCC plays 6C
Info: dddd plays 8D
Info: zzzzzzzz plays JS
S:
C:
D: K T
H: T
[C] play> Info: CCC won
Info: CCC plays 3D
Info: dddd plays AD
Info: zzzzzzzz plays KS
S:
C:
D: K
H: T
[D] play> Info: dddd won
Info: dddd plays QD
Info: zzzzzzzz plays AS
S:
C:
D:
H: T
[D] play> Info: CCC plays 6D
Info: 123456 won
Info: Team 1=220, Team 2=350
S: A K J 8 5 3 2
C: J 8 5
D:
H: Q 7 3
Bid> S: A K J 8 5 3 2
C: J 8 5
D:
H: Q 7 3
Lead> Info: CCC plays QS
Info: dddd plays TS
Info: zzzzzzzz plays 4S
Info: CCC won
Info: CCC plays 3C
Info: dddd plays AC
Info: zzzzzzzz plays KC
S: A K J 8 5 3
C: J 8 5
D:
H: Q 7 3
[C] play> Info: dddd won
Info: dddd plays KH
Info: zzzzzzzz plays 5H
S: A K J 8 5 3
C: J 8
D:
H: Q 7 3
[H] play> Info: CCC plays AH
Info: CCC won
Info: CCC plays 7S
Info: dddd plays 6S
Info: zzzzzzzz plays 9S
S: A K J 8 5 3
C: J 8
D:
H: Q 7
[S] play> Info: zzzzzzzz won
Info: zzzzzzzz plays AD
S: A K J 8 5
C: J 8
D:
H: Q 7
[D] play> Info: CCC plays 5D
Info: dddd plays JD
Info: zzzzzzzz won
Info: zzzzzzzz plays 9C
S: A K J 8 5
C: J
D:
H: Q 7
[C] play> Info: CCC plays 7C
Info: dddd plays TC
Info: 123456 won
S: A K J 8 5
C:
D:
H: Q 7
Lead> Info: CCC plays KD
Info: dddd plays 6C
Info: zzzzzzzz plays 8H
Info: zzzzzzzz won
Info: zzzzzzzz plays JH
S: K J 8 5
C:
D:
H: Q 7
[H] play> Info: CCC plays 2H
Info: dddd plays 4H
Info: 123456 won
S: K J 8 5
C:
D:
H: 7
Lead> Info: CCC plays 6H
Info: dddd plays 9H
Info: zzzzzzzz plays 2D
Info: dddd won
Info: dddd plays 2C
Info: zzzzzzzz plays 4C
S: K J 8 5
C:
D:
H:
[C] play> Info: CCC plays QC
Info: CCC won
Info: CCC plays 7D
Info: dddd plays 3D
Info: zzzzzzzz plays QD
S: K J 8
C:
D:
H:
[D] play> Info: zzzzzzzz won
Info: zzzzzzzz plays 8D
S: K J
C:
D:
H:
[D] play> Info: CCC plays TD
Info: dddd plays 9D
Info: CCC won
Info: CCC plays TH
Info: dddd plays 6D
Info: zzzzzzzz plays 4D
S: K
C:
D:
H:
[H] play> Info: CCC won
Info: Team 1=-80, Team 2=350
S: A K J 8 5 3 2
C: J T 5
D:
H: Q 7 3
Bid> Bid> S: A K J 8 5 3 2
C: J T 5
D:
H: Q 7 3
Lead> Info: CCC plays AH
Info: dddd plays 4H
Info: zzzzzzzz plays 5H
Info: CCC won
Info: CCC plays 7S
Info: dddd plays TS
Info: zzzzzzzz plays 4S
S: A K J 8 5 3 2
C: J T 5
D:
H: Q 7
[S] play> Info: 123456 won
S: K J 8 5 3 2
C: J T 5
D:
H: Q 7
Lead> Info: CCC plays QS
Info: dddd plays 6S
Info: zzzzzzzz plays 9S
Info: 123456 won
S: J 8 5 3 2
C: J T 5
D:
H: Q 7
Lead> Info: CCC plays QC
Info: dddd plays 2C
Info: zzzzzzzz plays 4C
Info: CCC won
Info: CCC plays KD
Info: dddd plays 3D
Info: zzzzzzzz plays 2D
S: J 8 5 3 2
C: J T
D:
H: Q 7
[D] play> Info: 123456 won
S: J 8 5 3 2
C: J T
D:
H: Q
Lead> Info: CCC plays TH
Info: dddd plays 9H
Info: zzzzzzzz plays 8H
Info: 123456 won
S: J 8 5 3 2
C: J T
D:
H:
Lead> Info: CCC plays 2H
Info: dddd plays AC
Info: zzzzzzzz plays 4D
Info: CCC won
Info: CCC plays 6H
Info: dddd plays KH
Info: zzzzzzzz plays JH
S: J 8 5 3
C: J T
D:
H:
[H] play> Info: dddd won
Info: dddd plays JD
Info: zzzzzzzz plays AD
S: J 8 5
C: J T
D:
H:
[D] play> Info: CCC plays 5D
Info: zzzzzzzz won
Info: zzzzzzzz plays 7C
S: J 8
C: J T
D:
H:
[C] play> Info: CCC plays 3C
Info: dddd plays 8C
Info: 123456 won
S: J 8
C: T
D:
H:
Lead> Info: CCC plays 9C
Info: dddd plays 6D
Info: zzzzzzzz plays KC
Info: 123456 won
S: J
C: T
D:
H:
Lead> Info: CCC plays 7D
Info: dddd plays 6C
Info: zzzzzzzz plays 8D
Info: 123456 won
S:
C: T
D:
H:
Lead> Info: CCC plays TD
Info: dddd plays 9D
Info: zzzzzzzz plays QD
Info: 123456 won
Info: Team 1=220, Team 2=350
S: Q 7
C: Q 7 3
D: K T 7 5
H: A T 6 2
Bid> S: Q 7
C: Q 7 3
D: K T 7 5
H: A T 6 2
Lead> S: Q 7
C: Q 7 3
D: K T 7 5
H: A T 6 2
Lead> Info: CCC plays KH
Info: dddd plays 5H
Info: zzzzzzzz plays 3H
Info: 123456 won
S: Q 7
C: Q 7 3
D: K T 7 5
H: T 6 2
Lead> Info: CCC plays JD
Info: dddd plays 2D
Info: zzzzzzzz plays 7H
Info: zzzzzzzz won
Info: zzzzzzzz plays QH
S: Q 7
C: Q 7 3
D: T 7 5
H: T 6 2
[H] play> Info: CCC plays 4H
Info: dddd plays 8H
Info: zzzzzzzz won
Info: zzzzzzzz plays AS
S: Q 7
C: Q 7 3
D: T 7 5
H: T 6
[S] play> Info: CCC plays 6S
Info: dddd plays 4S
Info: zzzzzzzz won
Info: zzzzzzzz plays KS
S: Q
C: Q 7 3
D: T 7 5
H: T 6
[S] play> S: Q
C: Q 7 3
D: T 7 5
H: T 6
[S] play> Info: CCC plays TS
Info: dddd plays 9S
Info: zzzzzzzz won
Info: zzzzzzzz plays JS
S:
C: Q 7 3
D: T 7 5
H: T 6
[S] play> Info: CCC plays AC
Info: dddd plays 4C
Info: 123456 won
S:
C: Q 7 3
D: T 7 5
H: T
Lead> Info: CCC plays TC
Info: dddd plays 9C
Info: zzzzzzzz plays 5C
Info: 123456 won
S:
C: 7 3
D: T 7 5
H: T
Lead> Info: CCC plays 2C
Info: dddd plays KC
Info: zzzzzzzz plays 8C
Info: dddd won
Info: dddd plays AD
Info: zzzzzzzz plays 8S
S:
C: 3
D: T 7 5
H: T
[D] play> Info: CCC plays 9D
Info: dddd won
Info: dddd plays JH
Info: zzzzzzzz plays 2S
S:
C: 3
D: 7 5
H: T
[H] play> S:
C: 3
D: 7 5
H: T
[H] play> Info: CCC plays 9H
Info: dddd won
Info: dddd plays 4D
Info: zzzzzzzz plays 3S
S:
C: 3
D: 7 5
H:
[D] play> Info: CCC plays 3D
Info: 123456 won
S:
C: 3
D: 7
H:
Lead> Info: CCC plays 6D
Info: dddd plays 8D
Info: zzzzzzzz plays JC
Info: dddd won
Info: dddd plays QD
Info: zzzzzzzz plays 5S
S:
C: 3
D:
H:
[D] play> Info: CCC plays 6C
Info: dddd won
Info: Team 1=520, Team 2=350
Info: Winner is Team 1
